packages/@aws-cdk/core/lib/deployment-environment.ts

```diff
@@ -0,0 +1,532 @@
+   */
+  readonly stagingBucketName?: string;
+
+  /**
+   * Name of the ECR repository to push Docker Images
+   *
+   * You must supply this if you have given a non-standard name to the ECR repository.
+   *
+   * The placeholders `${Qualifier}`, `${AWS::AccountId}` and `${AWS::Region}` will
+   * be replaced with the values of qualifier and the stack's account and region,
+   * respectively.
+   *
+   * @default "cdk-bootstrap-${Qualifier}-container-assets-${AWS::AccountId}-${AWS::Region}"
+   */
+  readonly ecrRepositoryName?: string;
+
+  /**
+   * The role to use to publish assets to this environment
+   *
+   * You must supply this if you have given a non-standard name to the publishing role.
+   *
+   * The placeholders `${Qualifier}`, `${AWS::AccountId}` and `${AWS::Region}` will
+   * be replaced with the values of qualifier and the stack's account and region,
+   * respectively.
+   *
+   * @default "cdk-bootstrap-publishing-role-${AWS::AccountId}-${AWS::Region}"
+   */
+  readonly assetPublishingRoleName?: string;
+
+  /**
+   * External ID to use when assuming role for asset publishing
+   *
+   * @default - No external ID
+   */
+  readonly assetPublishingExternalId?: string;
+
+  /**
+   * The role to assume to execute
+   *
+   * You must supply this if you have given a non-standard name to the publishing role.
+   *
+   * The placeholders `${Qualifier}`, `${AWS::AccountId}` and `${AWS::Region}` will
+   * be replaced with the values of qualifier and the stack's account and region,
+   * respectively.
+   *
+   * @default "cdk-bootstrap-deploy-action-role-${AWS::AccountId}-${AWS::Region}"
+   */
+  readonly deployActionRoleName?: string;
+
+  /**
+   * The role CloudFormation will assume when deploying the Stack
+   *
+   * You must supply this if you have given a non-standard name to the execution role.
+   *
+   * The placeholders `${Qualifier}`, `${AWS::AccountId}` and `${AWS::Region}` will
+   * be replaced with the values of qualifier and the stack's account and region,
+   * respectively.
+   *
+   * @default "cdk-bootstrap-cfn-exec-role-${AWS::AccountId}-${AWS::Region}"
+   */
+  readonly cloudFormationExecutionRole?: string;
+
+  /**
+   * Qualifier to disambiguate multiple environments in the same account
+   *
+   * You can use this and leave the other naming properties empty if you have deployed
+   * the bootstrap environment with standard names but only differnet qualifiers.
+   *
+   * @default "hnb659fds"
+   */
+  readonly qualifier?: string;
+}
+
+/**
+ * Uses conventionally named roles and reify asset storage locations
+ *
+ * This DeploymentEnvironment is the only DeploymentEnvironment that generates
+ * an asset manifest, and is required to deploy CDK applications using the
+ * `@aws-cdk/app-delivery` CI/CD library.
+ *
+ * Requires the environment to have been bootstrapped with Bootstrap Stack V2.
+ */
+export class ConventionModeDeploymentEnvironment implements IDeploymentEnvironment {
```

Comment:
I don't feel super comfortable with this name. How about just `DefaultEnvironmentProvider` (given we rename this to `IDeploymentEnvironment`)?